### ISSUE TITLE
[Possible Bug?] Newer Versions Lead to Segmentation Fault under Certain Conditions on Ubuntu22.04

### ISSUE THREAD

**Issue by Zephyr69** [pooya-mohammadi/deep_utils#51]
System: Ubuntu22.04lts
Task: Starting to train a yolov5 model with deep_utils being one of the imported modules (that doesn't do anything during training).
Symptom: When trying to import deep_utils, an error message is thrown before any other console output: Process finished with exit code 139 (interrupted by signal 11: SIGSEGV).

Package                      Version
---------------------------- ------------------------
absl-py                      1.4.0
albumentations               1.3.0
apturl                       0.5.2
astunparse                   1.6.3
bcrypt                       3.2.0
blinker                      1.4
Brlapi                       0.8.3
cachetools                   5.3.0
certifi                      2020.6.20
chardet                      4.0.0
charset-normalizer           3.0.1
click                        8.0.3
cmake                        3.25.2
colorama                     0.4.4
command-not-found            0.3
contourpy                    1.0.7
cryptography                 3.4.8
cupshelpers                  1.0
cvms                         0.1.1
cycler                       0.11.0
dbus-python                  1.2.18
deep-utils                   0.9.5
defer                        1.0.6
distro                       1.7.0
distro-info                  1.1build1
duplicity                    0.8.21
fasteners                    0.14.1
filelock                     3.9.0
flatbuffers                  23.3.3
fonttools                    4.39.0
future                       0.18.2
gast                         0.5.3
google-auth                  2.16.2
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
grpcio                       1.51.3
h5py                         3.8.0
httplib2                     0.20.2
idna                         3.3
imageio                      2.26.0
importlib-metadata           4.6.4
jeepney                      0.7.1
joblib                       1.2.0
keras                        2.8.0
Keras-Preprocessing          1.1.2
keyring                      23.5.0
kiwisolver                   1.4.4
language-selector            0.1
launchpadlib                 1.10.16
lazr.restfulclient           0.14.4
lazr.uri                     1.0.6
lazy_loader                  0.1
libclang                     15.0.6.1
lit                          15.0.7
lockfile                     0.12.2
louis                        3.20.0
macaroonbakery               1.3.1
Mako                         1.1.3
Markdown                     3.4.1
MarkupSafe                   2.1.2
matplotlib                   3.7.1
monotonic                    1.6
more-itertools               8.10.0
mpmath                       1.2.1
netifaces                    0.11.0
networkx                     3.0
numpy                        1.24.2
oauthlib                     3.2.0
olefile                      0.46
opencv-python                4.5.5.64
opencv-python-headless       4.7.0.72
opt-einsum                   3.3.0
packaging                    23.0
pandas                       1.5.3
paramiko                     2.9.3
pexpect                      4.8.0
Pillow                       9.0.1
pip                          22.0.2
protobuf                     3.12.4
psutil                       5.9.0
ptyprocess                   0.7.0
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycairo                      1.20.1
pycups                       2.0.1
PyGObject                    3.42.1
PyJWT                        2.3.0
pymacaroons                  0.13.0
PyNaCl                       1.5.0
pyparsing                    2.4.7
pyRFC3339                    1.1
python-apt                   2.4.0+ubuntu1
python-dateutil              2.8.1
python-debian                0.1.43ubuntu1
pytorch-triton               2.0.0+b8b470bc59
pytz                         2022.1
PyWavelets                   1.4.1
pyxdg                        0.27
PyYAML                       5.4.1
qudida                       0.0.4
reportlab                    3.6.8
requests                     2.28.2
requests-oauthlib            1.3.1
rsa                          4.9
scikit-image                 0.20.0
scikit-learn                 1.2.1
scipy                        1.10.1
screen-resolution-extra      0.0.0
seaborn                      0.12.2
SecretStorage                3.3.1
setuptools                   59.6.0
six                          1.16.0
ssh-import-id                5.11
sympy                        1.11.1
systemd-python               234
tensorboard                  2.8.0
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.8.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.2.0
tf-estimator-nightly         2.8.0.dev2021122109
thop                         0.1.1.post2209072238
threadpoolctl                3.1.0
tifffile                     2023.2.28
timm                         0.6.7
torch                        2.0.0.dev20230301+cu118
torchaudio                   2.0.0.dev20230305+cu118
torchvision                  0.15.0.dev20230305+cu118
tqdm                         4.64.0
typing_extensions            4.5.0
ubuntu-advantage-tools       8001
ubuntu-drivers-common        0.0.0
ufw                          0.36.1
unattended-upgrades          0.1
urllib3                      1.26.5
usb-creator                  0.3.7
wadllib                      1.3.6
Werkzeug                     2.2.3
wheel                        0.37.1
wrapt                        1.15.0
xdg                          5
xkit                         0.0.0
zipp                         1.0.0

The last usable version is 0.9.5.
